include/openssl/engine.h

```diff
@@ -331,10 +331,14 @@ OSSL_DEPRECATEDIN_3_0 int ENGINE_add(ENGINE *e);
 #  ifndef OPENSSL_NO_DEPRECATED_3_0
 OSSL_DEPRECATEDIN_3_0 int ENGINE_remove(ENGINE *e);
 #  endif
-/* Retrieve an engine from the list by its unique "id" value. */
+/* Retrieve an engine from the list by its unique "id", may load new engines. */
 #  ifndef OPENSSL_NO_DEPRECATED_3_0
 OSSL_DEPRECATEDIN_3_0 ENGINE *ENGINE_by_id(const char *id);
 #  endif
+/* Retrieve an engine from the list by its unique "id" value. */
+#  ifndef OPENSSL_NO_DEPRECATED_3_0
+OSSL_DEPRECATEDIN_3_0 ENGINE *ENGINE_search(const char *id);
```

Comment:
You are adding a method that is directly deprecated ... 
do you really need it ?
No sure this change will be accepted, as ENGINE API is nowadays deprecated, to prepare its future removal.

Reply:
As you are modifying a public header file, you need to run a `make update` , and commit its results

Reply:
I was told in #17025 to move the check inside eng_dyn.c and to implement the engine_list_search() for it. If there is a way to do this without exposing a new API function I would gladly do it.

Please keep in mind that as mentioned in #17023, the issue is relevant to existing branches (1.1.1) and should be reviewed as such.

Reply:
Also added the results of make update as part of the commit

Reply:
Can't this be just an internal function?

Reply:
I mean, move this declaration to eng_local.h

Reply:
I'll try it, wasn't familiar with eng_local.h

Reply:
ok, and please make this lower-case engine_search, to make it clear that it is not gong to be exported.

Reply:
Sure, just did. Uploading it now